cpp/src/parquet/properties.h

```diff
@@ -535,6 +535,19 @@ static constexpr bool kArrowDefaultUseThreads = false;
 // Default number of rows to read when using ::arrow::RecordBatchReader
 static constexpr int64_t kArrowDefaultBatchSize = 64 * 1024;
 
+// At times the arrow code base might contain multiple iterations
+// of code to read/write data to Parquet.  This enum allows
+// selection of the given code.
+enum class ArrowEngineVersion {
+  // The current default engine. This doesn't support mixed
+  // struct list date.
+  kDefault = 0,
+
+  // Not fully implemented.  When done this will support reading/
+  // writing all forms of nested data.
+  kNextGen = 1
```

Comment:
Will number 2 be named `kNextNextGen`? :-) I suggest we find other names (perhaps simply version names: `v1`, `v2`).

(also, I don't think the `k` prefix is necessary if it's a qualified enum member?)

Reply:
I think qualified enums are supposed to be treated like [constants](https://google.github.io/styleguide/cppguide.html#Enumerator_Names).  In this case I changed to V1 and V2